client/Makefile

```diff
@@ -34,13 +34,17 @@ start-frontend:
 
 .PHONY: e2e
 e2e:
-	node node_modules/jest/bin/jest.js \
+	JEST_ENV=${PROD} \
+	CXG_CLIENT_PORT=${CXG_SERVER_PORT} \
```

Comment:
tl;dr: I think you can delete the addition of these environment variables in `e2e` and `e2e-annotations`. Details follow:

* The intent behind `common.mk` was that it introduced defaults from one source of truth, `environment.default`. Having overriding env vars in the code and Makefiles kinda breaks that convention. This might be OK, I'm less zealous about there being more than one source of truth on this that I used to be.
* You could make `JEST_ENV=prod` in `environment.default` and delete the variable `PROD` there and `JEST_ENV` here. Cleaner and has the same effect 🙂
* Switching `CXG_CLIENT_PORT` and `CXG_SERVER_PORT` makes my head spin a bit. Our naming convention isn't great, but I think we should stick to `CLIENT` referring to the frontend and `SERVER` referring to the backend API.

Reply:
Thanks for the great feedback and background on this, @mweiden ! I see what you mean now 😄 

This idea just came to me, but what's your thought on converting `environment.default` to `environment.default.json`, so both Python and JS can consume it, and we will have that as the single source of truth?

Re: `CXG_CLIENT_PORT` and `CXG_SERVER_PORT`: 

Yeah the only reason I had to do this is that GHA's e2e tests hit pages at `CXG_SERVER_PORT`(5005), but FE local development hit `CXG_CLIENT_PORT` (3000) (cuz we spin up `make start-frontend`).

So I decided to update `config.js` with:

```ts
export const CXG_CLIENT_PORT = 3000;
export const appPort = process.env.CXG_CLIENT_PORT || CXG_CLIENT_PORT;
```

So e2e tests will hit 3000 by default, and CI/CD to hit 5005 through passing `CXG_CLIENT_PORT=${CXG_SERVER_PORT}` in the make file:

```
JEST_ENV=${PROD} \
	CXG_CLIENT_PORT=${CXG_SERVER_PORT} \
		node node_modules/jest/bin/jest.js \
		--verbose false \
		--config __tests__/e2e/e2eJestConfig.json \
		e2e/e2e.test.js
```

Reply:
Ah just realized that it's not about sharing environment defaults between JS and Python, but between JS and Makefile 😆 

In that case, maybe this would do?

```
define GetValueFromJson
$(shell node -p '\
    const getVal = (key = "", obj = {}) => {
          const currKey = key.split(".")[0];
          const val = obj[currKey];
          if(typeof val !== "object") return val;
          const nextKey = key.split(".").slice(1).join(".");
          return getVal(nextKey, val);
    }; \
    getVal(`$(1)`.replace(" ", ""), require("./package.json")); \
')
endef

PORT := $(call GetValueFromJson, config.port)
```

[source](https://stackoverflow.com/questions/36664010/read-json-and-assign-to-a-list-of-make-variables)

Reply:
Just updated the PR to use `environment.default.json`!